### PR TITLE
Update French translations for v2023.1

### LINKED ISSUE #1229: Update translation to French (Engine)

### PR DESCRIPTION
**Pull request by marc-vanderwal**
## Purpose

This PR updates the French translation of zonemaster-engine in preparation of release v2023.1.

## Context

Fixes #1229.

## Changes

Update French translations.

## How to test this PR

Run a test and see if new translations are applied.

### CODE REVIEW

File: share/fr.po [hunk 1]
```diff
@@ -157,14 +157,69 @@ msgstr "Le nom de domaine doit avoir au moins un serveur de nom fonctionnel"
...
+#, perl-brace-format
+msgid "Name server \"{nsname}\" cannot be resolved into an IP address."
+msgstr ""
+"Le serveur de noms \"{nsname}\" ne peut pas être résolu en une adresse IP."
+
+#. BASIC:B02_NS_NO_RESPONSE
+#, perl-brace-format
+msgid "Name server \"{ns}\" does not respond to an SOA query."
+msgstr ""
+"Le serveur de noms \"{ns}\" ne répond pas aux requêtes de type \"SOA\"."
```
[the remark this answers is not in the document]
Comment (marc-vanderwal) [reply share/fr.po:212]: Currently the translation is rather inconsistent on the use of double quotes around the resource record types. Might as well use the opportunity to change that. I agree with you by thinking that they should best be avoided.

File: share/fr.po [hunk 2]
```diff
@@ -491,6 +525,70 @@ msgstr ""
...
+msgid "Prefix database returned no information for IP address {ns_ip}."
+msgstr ""
+"La base de données des préfixes IP ne retourne aucune information pour "
+"l'adresse IP {ns_ip}."
+
+#. CONNECTIVITY:CN04_ERROR_PREFIX_DATABASE
+#, perl-brace-format
+msgid "Prefix database error. No data to analyze for IP address {ns_ip}."
+msgstr ""
+"La base de données des préfixes IP retourne une erreur. Aucune donnée à "
```
[the remark this answers is not in the document]
Comment (marc-vanderwal) [reply share/fr.po:553]: Good idea. The same could be said for the current translation for ERROR_ASN_DATABASE, by the way.

File: share/fr.po [hunk 3]
```diff
@@ -2323,6 +2409,10 @@ msgstr "Test d'une réponse tronquée pour une requête EDNS"
 msgid "Test for unknown version with unknown OPTION-CODE"
 msgstr "Test de numéro de version inconnue avec code d’option inconnu"
 
+#. NAMESERVER:NAMESERVER15
+msgid "Checking for revealed software version"
+msgstr "Test de version logicielle révélée"
```
[the remark this answers is not in the document]
Comment (marc-vanderwal) [reply share/fr.po:2414]: Me neither, actually. I like your suggestion.

File: share/fr.po [hunk 4]
```diff
@@ -157,14 +157,69 @@ msgstr "Le nom de domaine doit avoir au moins un serveur de nom fonctionnel"
...
+#, perl-brace-format
+msgid "Broken response from name server \"{ns}\" on an SOA query."
+msgstr "Réponse cassée du serveur de noms \"{ns}\" à une requête de type SOA."
+
+#. BASIC:B02_NS_NOT_AUTH
+#, perl-brace-format
+msgid ""
+"Name server \"{ns}\" does not give an authoritative answer on an SOA query."
+msgstr ""
+"La réponse du serveur de noms {ns} à une requête de type \"SOA\" ne fait pas "
```
[the remark this answers is not in the document]
Comment (matsduf) [reply share/fr.po]: In the msgid it is not consistent what is quoted what is not or when an element is quoted and when it is not. I am not sure if it is consistent between `" "` and `' '`, respectively.

I am not sure that the target language must have the same rules as in msgid, but we should try to be consistent. That is, however, a gigantic effort.

File: share/fr.po [hunk 5]
```diff
@@ -3292,10 +3402,11 @@ msgstr ""
 #. ZONE:Z09_UNEXPECTED_RCODE_MX
 #, perl-brace-format
 msgid ""
-"Unexpected RCODE value on the MX query from name servers \"{ns_ip_list}\"."
+"Unexpected RCODE value ({rcode}) on the MX query from name servers "
+"\"{ns_ip_list}\"."
 msgstr ""
-"Les serveurs de noms \"{ns_ip_list}\" ont répondu à une requête de type \"MX"
-"\" avec un code retour inattendu."
+"Code de retour inattendu ({rcode}) en réponse à une requête de type \"MX\" "
```
[the remark this answers is not in the document]
Comment (marc-vanderwal) [reply share/fr.po]: My bad, I fixed it in a different commit on another branch. Fixing it here.

**Review by matsduf — APPROVED** [review]
Everything I can see looks good.

**Pull request merged**